File: src/main/javaDDD/com/template/infrastructure/repositories/PostgresqlDBUserRepository.java
package com.template.infrastructure.repositories;

import org.springframework.stereotype.Repository;
import org.springframework.data.jpa.repository.JpaRepository;
import com.template.domain.models.User;
import java.util.List;
import java.util.Optional;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.context.annotation.Primary;
import org.springframework.stereotype.Component;

import com.template.domain.repositories.UserRepository;

@Component
@Primary
public class PostgresqlDBUserRepository implements UserRepository {
    private final JpaUserRepository userRepository;

    @Autowired
    public PostgresqlDBUserRepository(final JpaUserRepository userRepository) {
        this.userRepository = userRepository;
    }

    @Override
    public Optional<User> findById(final Long id) {
        return userRepository.findById(id);
    }

    @Override
    public User save(final User user) {
        return userRepository.save(user);
    }

    @Override
    public List<User> findAll() {
        return userRepository.findAll();
    }

    @Override
    public void deleteById(Long id) {
        userRepository.deleteById(id);
    }
}

File: src/main/kotlinDDD/com/template/infrastructure/repositories/PostgresqlDBUserRepository.kt
package com.template.infrastructure.repositories

import org.springframework.stereotype.Repository
import org.springframework.data.jpa.repository.JpaRepository
import com.template.domain.models.User
import kotlin.collections.MutableList
import java.util.Optional
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.context.annotation.Primary
import org.springframework.stereotype.Component
import com.template.domain.repositories.UserRepository

@Component
@Primary
class PostgresqlDBUserRepository @Autowired constructor(userRepository: JpaUserRepository) : UserRepository {
    private val userRepository: JpaUserRepository

    init {
        this.userRepository = userRepository
    }

    override fun findById(id: Long): User? {
        return when(val user = userRepository.findById(id)) {
            Optional.ofNullable(null) -> {
                return null
            }
            else -> {
                return user.get()
            }
        }
    }

    override fun save(user: User): User? {
        return userRepository.save(user)
    }

    override fun findAll(): MutableList<User?> {
        return userRepository.findAll()
    }

    override fun deleteById(id: Long) {
        userRepository.deleteById(id)
    }
}
